exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,3725 @@
+	"Rfq":                                    `{"params" : {"data" : {"state" : true,"side" : null,"last_rfq_tstamp" : 1634816143836,"instrument_name" : "BTC-PERPETUAL","amount" : null	  },"channel" : "rfq.btc"	},	"method" : "subscription",	"jsonrpc" : "2.0"  }`,
+	"Instrument Trades":                      `{"params":{"data":[{"trade_seq":30289442,"trade_id" : "48079269","timestamp" : 1590484512188,"tick_direction" : 2,"price" : 8950,"mark_price" : 8948.9,"instrument_name" : "BTC-PERPETUAL","index_price" : 8955.88,"direction" : "sell","amount" : 10}],"channel" : "trades.BTC-PERPETUAL.raw"},"method":"subscription","jsonrpc":"2.0"}`,
+	"Instruments Ticker":                     `{"params" : {"data" : {"timestamp" : 1623060194301,"stats" : {"volume_usd" : 284061480,"volume" : 7871.02139035,"price_change" : 0.7229,"low" : 35213.5,"high" : 36824.5},"state" : "open","settlement_price" : 36169.49,"open_interest" : 502097590,"min_price" : 35898.37,"max_price" : 36991.72,"mark_price" : 36446.51,"last_price" : 36457.5,"interest_value" : 1.7362511643080387,"instrument_name" : "BTC-PERPETUAL","index_price" : 36441.64,"funding_8h" : 0.0000211,"estimated_delivery_price" : 36441.64,"current_funding" : 0,"best_bid_price" : 36442.5,"best_bid_amount" : 5000,"best_ask_price" : 36443,"best_ask_amount" : 100	  },	  "channel" : "ticker.BTC-PERPETUAL.raw"	},	"method" : "subscription",	"jsonrpc" : "2.0"  }`,
+}
+
+func TestProcessPushData(t *testing.T) {
+	t.Parallel()
+	for x := range websocketPushData {
+		err := d.wsHandleData([]byte(websocketPushData[x]))
+		require.NoError(t, err)
+	}
+}
+
+func TestFormatFuturesTradablePair(t *testing.T) {
+	t.Parallel()
+	futuresInstrumentsOutputList := map[currency.Pair]string{
+		{Delimiter: currency.DashDelimiter, Base: currency.BTC, Quote: currency.NewCode("PERPETUAL")}:                   "BTC-PERPETUAL",
+		{Delimiter: currency.DashDelimiter, Base: currency.AVAX, Quote: currency.NewCode("USDC-PERPETUAL")}:             "AVAX_USDC-PERPETUAL",
+		{Delimiter: currency.DashDelimiter, Base: currency.ETH, Quote: currency.NewCode("30DEC22")}:                     "ETH-30DEC22",
+		{Delimiter: currency.DashDelimiter, Base: currency.SOL, Quote: currency.NewCode("30DEC22")}:                     "SOL-30DEC22",
+		{Delimiter: currency.DashDelimiter, Base: currency.NewCode("BTCDVOL"), Quote: currency.NewCode("USDC-28JUN23")}: "BTCDVOL_USDC-28JUN23",
+	}
+	for pair, instrumentID := range futuresInstrumentsOutputList {
+		instrument := d.formatFuturesTradablePair(pair)
+		require.Equal(t, instrumentID, instrument)
+	}
+}
+
+func TestWSRetrieveCombos(t *testing.T) {
+	t.Parallel()
+	sleepUntilTradablePairsUpdated()
+	_, err := d.WSRetrieveCombos(currency.EMPTYCODE)
+	require.ErrorIs(t, err, currency.ErrCurrencyCodeEmpty)
+	_, err = d.WSRetrieveCombos(futureComboTradablePair.Base)
+	assert.NoError(t, err)
+}
+
+func instantiateTradablePairs() {
```

Comment:
Can be cleaned up to:

````
func instantiateTradablePairs() {
	d.Websocket.Wg.Add(1)
	go func(tpfChan chan struct{}) {
		defer d.Websocket.Wg.Done()

		handleError := func(err error, msg string) {
			if err != nil {
				close(tpfChan)
				log.Fatalf("%s: %v", msg, err)
			}
		}

		assets := []asset.Item{asset.Options, asset.OptionCombo, asset.FutureCombo}
		for _, assetType := range assets {
			pairs, err := d.FetchTradablePairs(context.Background(), assetType)
			handleError(err, fmt.Sprintf("Error fetching tradable pairs for asset type %v", assetType))

			err = d.UpdatePairs(pairs, assetType, false, true)
			handleError(err, fmt.Sprintf("Error updating tradable pairs for asset type %v", assetType))
		}

		updateTradablePair := func(assetType asset.Item, tradablePair *currency.Pair) {
			if d.CurrencyPairs.IsAssetEnabled(assetType) == nil {
				pairs, err := d.GetEnabledPairs(assetType)
				handleError(err, fmt.Sprintf("Failed to get enabled pairs for asset type %v", assetType))

				if len(pairs) == 0 {
					handleError(currency.ErrCurrencyPairsEmpty, fmt.Sprintf("No enabled pairs for asset type %v", assetType))
				}

				if assetType == asset.Options {
					*tradablePair, err = d.FormatExchangeCurrency(pairs[0], assetType)
					handleError(err, "Failed to format exchange currency for options pair")
				} else {
					*tradablePair = pairs[0]
				}
			}
		}

		updateTradablePair(asset.Options, &optionsTradablePair)
		updateTradablePair(asset.OptionCombo, &optionComboTradablePair)
		updateTradablePair(asset.FutureCombo, &futureComboTradablePair)

		tradablePairsFetchedStatusLock.Lock()
		tradablePairsFetched = true
		tradablePairsFetchedStatusLock.Unlock()
		close(tpfChan)
	}(fetchTradablePairChan)
}
```

Or alternatively you can UpdateTradablePairs for all assets, then set your options/option combo and futures combo pair vars

Reply:
This will time compared with the previous one. So, I'll go with this one. Thanks!